File: CS/WinSample.Module.Win/ViewController1.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.Data.Filtering;

namespace WinSample.Module.Win {
    public partial class ViewController1 : ViewController {
        public ViewController1() {
            InitializeComponent();
            RegisterActions(components);
            TargetViewId = "BO1_ListView";
        }
        protected override void OnActivated() {
            base.OnActivated();
            simpleAction1.Checked = false;
        }
        private const string filterValue = "Not IsNull([Property1])";
        private const string filterName = "ListViewFilter";
        private void simpleAction1_Execute(object sender, SimpleActionExecuteEventArgs e) {
            if (simpleAction1.Checked) {
                ((ListView)View).CollectionSource.Criteria[filterName] = CriteriaOperator.Parse(filterValue);
            } else {
                ((ListView)View).CollectionSource.Criteria.Remove(filterName);
            }
        }
    }
}

File: VB/WinSample.Module.Win/ViewController1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.Data.Filtering

Namespace WinSample.Module.Win
	Partial Public Class ViewController1
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
			TargetViewId = "BO1_ListView"
		End Sub
		Protected Overrides Overloads Sub OnActivated()
			MyBase.OnActivated()
			simpleAction1.Checked = False
		End Sub
		Private Const filterValue As String = "Not IsNull([Property1])"
		Private Const filterName As String = "ListViewFilter"
		Private Sub simpleAction1_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs) Handles simpleAction1.Execute
			If simpleAction1.Checked Then
				CType(View, ListView).CollectionSource.Criteria(filterName) = CriteriaOperator.Parse(filterValue)
			Else
				CType(View, ListView).CollectionSource.Criteria.Remove(filterName)
			End If
		End Sub
	End Class
End Namespace
